src/System.Runtime/tests/System/LazyTests.cs

```diff
@@ -578,5 +579,55 @@ private struct LIX
             public override int GetHashCode() => f.GetHashCode();
             public override string ToString() => "LIX<" + f + ">";
         }
+
+        private readonly static Random s_randomizer = new Random();
+
+        private static ClassWithInitFunction s_refInitFunctionLockedObject;
+
+        public static void EnsureInitialized_RefInitFunctionLocked()
+        {
+            const int MaxCount = 2000;
+
+            List<Task> threads = new List<Task>();
+            ClassWithInitFunction.InitCounters();
+            Object lockObject = new object();
+
+            for (int count = 0; count < MaxCount; count++)
+            {
+                threads.Add(Task.Run(() => LazyInitializer.EnsureInitialized(ref s_refInitFunctionLockedObject, ref lockObject, () => ClassWithInitFunction.InitFunction())));
+            }
+            Task.WaitAll(threads.ToArray());
+            Assert.Equal(1, ClassWithInitFunction.CountInitialized);
+        }
+
+        private class ClassBase
```

Comment:
I had created this as common base with counters and sleep function to be used for ctor based API overload test case. Now that is removed, but I missed merging this into the derived class...